app/models/manageiq/providers/ansible_tower/shared/inventory/parser/automation_manager.rb

```diff
@@ -109,4 +111,29 @@ def credentials
       end
     end
   end
+
+  def workflows
+    collector.workflow_job_templates.each do |job_template|
+      inventory_object = persister.workflows.find_or_build(job_template.id.to_s)
+      inventory_object.description = job_template.description
+      inventory_object.name = job_template.name
+      inventory_object.survey_spec = job_template.survey_spec_hash
+      inventory_object.variables = job_template.extra_vars_hash
+    end
+  end
+
+  def workflow_nodes
+    collector.workflow_job_template_nodes.each do |node|
+      inventory_object = persister.workflow_nodes.find_or_build(node.id.to_s)
+      inventory_object.workflow = persister.workflows.lazy_find(node.workflow_job_template_id.to_s)
+      inventory_object.configuration_script = persister.configuration_scripts.lazy_find(node.unified_job_template_id)
+      %w(success failure always).each do |condition|
+        node.send("#{condition}_nodes_id").each do |child_id|
+          child_node = persister.workflow_nodes.lazy_find(child_id)
+          # child_node.parent = inventory_object
```

Comment:
@Ladas How do I get Line 133-134 to work? 
or @agrare ?

Reply:
hm, doing this from this direction, you could just

```ruby
child_node = persister.workflow_nodes.find_or_build(child_id)
child_node.parent = inventory_object
```

then the upper find_or_build should fill the rest of the info, when it reaches the node. How is the parent implemented? Is it just the foreign key?

Reply:
Great! Thanks @Ladas